CORE-15757  flow session refactor

LWogan opened the pull request:
Flow Session Refactor to reduce the total messages exchanged between session counterparties. 

SessionInit info can now be piggybacked onto SessionData messages.

SessionAcks are removed.

SessionContextProperties are piggybacked onto any SessionEvents as SessionConfirm is not always sent.

Add new initiateFlow API to allow the initiator to define whether a SessionClose message should be sent by the initiated party when their flow finishes or when they call close. Whichever comes first.

Runtime-OS:
https://github.com/corda/corda-runtime-os/pull/4446
E2E Tests
https://github.com/corda/corda-e2e-tests/pull/218

LWogan committed:
CORE-15747 Initial schema changes to support initial session refactor

LWogan committed:
CORE-15747 Initial schema changes to support initial session refactor

LWogan committed:
CORE-15747 Initial schema changes to support initial session refactor

chrisr3 commented on application/src/main/java/net/corda/v5/application/messaging/FlowMessaging.java:
```diff
@@ -122,7 +147,7 @@ public interface FlowMessaging {
     @Suspendable
     @NotNull
-    FlowSession initiateFlow(@NotNull MemberX500Name x500Name, @NotNull FlowContextPropertiesBuilder flowContextPropertiesBuilder);
+    FlowSession initiateFlow(@NotNull MemberX500Name x500Name, @NotNull Boolean requireClose, @NotNull FlowContextPropertiesBuilder flowContextPropertiesBuilder);
```
By `@NotNull Boolean` I assume you really mean `boolean`?

chrisr3 commented on application/src/main/java/net/corda/v5/application/messaging/FlowMessaging.java:
```diff
@@ -84,6 +84,28 @@ public interface FlowMessaging {
+     */
+    @Suspendable
+    @NotNull
+    FlowSession initiateFlow(@NotNull MemberX500Name x500Name, Boolean requireClose);
```
Also `boolean` rather than `Boolean`?

LWogan committed:
CORE-15747 Reinstate sessioninit as it wil be used for counterparty flow info and flows which start with a receive

LWogan committed:
CORE-15747 make SessionInit nullable

LWogan commented:
@chrisr3 as mentioned in the draft text this PR is not for review yet

LWogan committed:
CORE-15747 boolean not Boolean

chrisr3 commented:
> @chrisr3 as mentioned in the draft text this PR is not for review yet

This change looks like [something unexpected that is difficult just to ignore](https://www.youtube.com/watch?v=d4HjDc44_VQ)...

LWogan commented:
> > @chrisr3 as mentioned in the draft text this PR is not for review yet
> 
> This change looks like [something unexpected that is difficult just to ignore](https://www.youtube.com/watch?v=d4HjDc44_VQ)...

I may have made some updates this morning.... completely unrelated to your comments.... 👀

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
CORE-14523 flow worker now consumes from p2p locally hosted identity topic (#1213)

change topic permissions

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
CORE-16174 move session context props from SessionInit to SessionEvent so it can be set both directions (#1212)

Remove config fields not used anymore.
counterparty lookups are no longer retriable.
Session events are not resent.

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
CORE-15757 reenable tests

LWogan committed:
CORE-15757 add initiate flow api to preserve backwards compatability

LWogan committed:
CORE-15757 add initiate flow api to preserve backwards compatability

LWogan committed:
CORE-15757 add initiate flow api to preserve backwards compatability

ben-millar commented on application/src/test/java/net/corda/v5/application/messaging/FlowMessagingJavaApiTest.java:
```diff
@@ -4,8 +4,7 @@
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.*;
```
Minor: wildcard import

LWogan committed:
CORE-15757 fix wildcard import

LWogan committed:
CORE-15757 update after merge

LWogan committed:
CORE-15757 update after merge
